crates/router/src/core/payments/flows/setup_mandate_flow.rs

```diff
@@ -63,6 +63,21 @@ impl Feature<api::SetupMandate, types::SetupMandateRequestData> for types::Setup
         connector_request: Option<services::Request>,
         key_store: &domain::MerchantKeyStore,
     ) -> RouterResult<Self> {
+        if self
+            .request
+            .customer_acceptance
+            .clone()
+            .or(self
+                .request
+                .setup_mandate_details
+                .as_ref()
+                .and_then(|mandate_data| mandate_data.customer_acceptance.clone()))
+            .is_none()
```

Comment:
Can we have this validation in `GetTrackers` instead? `decide_flows` is called after `update_trackers`, which usually sets the payment intent and attempt statuses to a variant of `Processing`. It doesn't make much sense to deny making a payment from our end if we have already updated the statuses as so.